experimental/packages/otlp-transformer/src/common/types.ts

```diff
@@ -14,72 +14,71 @@
  * limitations under the License.
  */
 
-/** Properties of an InstrumentationScope. */
-export interface IInstrumentationScope {
-  /** InstrumentationScope name */
-  name: string;
-
-  /** InstrumentationScope version */
-  version?: string;
-
-  /** InstrumentationScope attributes */
-  attributes?: IKeyValue[];
-
-  /** InstrumentationScope droppedAttributesCount */
-  droppedAttributesCount?: number;
-}
-
-/** Properties of a KeyValue. */
-export interface IKeyValue {
-  /** KeyValue key */
-  key: string;
-
-  /** KeyValue value */
-  value: IAnyValue;
-}
-
-/** Properties of an AnyValue. */
-export interface IAnyValue {
-  /** AnyValue stringValue */
-  stringValue?: string | null;
-
-  /** AnyValue boolValue */
-  boolValue?: boolean | null;
-
-  /** AnyValue intValue */
-  intValue?: number | null;
-
-  /** AnyValue doubleValue */
-  doubleValue?: number | null;
-
-  /** AnyValue arrayValue */
-  arrayValue?: IArrayValue;
-
-  /** AnyValue kvlistValue */
-  kvlistValue?: IKeyValueList;
-
-  /** AnyValue bytesValue */
-  bytesValue?: Uint8Array;
-}
-
-/** Properties of an ArrayValue. */
-export interface IArrayValue {
-  /** ArrayValue values */
-  values: IAnyValue[];
-}
-
-/** Properties of a KeyValueList. */
-export interface IKeyValueList {
-  /** KeyValueList values */
-  values: IKeyValue[];
-}
-
-export interface LongBits {
-  low: number;
-  high: number;
-}
-
-export type Fixed64 = LongBits | string | number;
+import { Type, type Static } from "@sinclair/typebox";
+
+export const OtelCommonTypes = Type.Module({
+  ResourceAttributes: Type.Object({
+    key: Type.String(),
+    value: Type.Any(),
+  }),
+  IResource: Type.Object({
+    attributes: Type.Array(Type.Ref("ResourceAttributes")),
+  }),
+  IKeyValue: Type.Object({
+    key: Type.String(),
+    value: Type.Ref("IAnyValue"),
+  }),
+  IKeyValueList: Type.Object({
+    values: Type.Array(Type.Ref("IKeyValue")),
+  }),
+  IArrayValue: Type.Object({
+    values: Type.Array(Type.Ref("IAnyValue")),
+  }),
+  IAnyValue: Type.Object({
+    stringValue: Type.Optional(Type.Union([Type.String(), Type.Null()])),
+    boolValue: Type.Optional(Type.Union([Type.Boolean(), Type.Null()])),
+    intValue: Type.Optional(Type.Union([Type.Number(), Type.Null()])),
+    doubleValue: Type.Optional(Type.Union([Type.Number(), Type.Null()])),
+    arrayValue: Type.Optional(Type.Ref("IArrayValue")),
+    kvlistValue: Type.Optional(Type.Ref("IKeyValueList")),
+    bytesValue: Type.Optional(Type.Uint8Array()),
+  }),
```

Comment:
The original type for this before my PR was given as an intersection of possible keys

However, possibly this was meant to be a union? Is it possible for multiple of these keys to be present, or is it always at most one of them?

Reply:
It looks like, given the `toAnyValue`, this is indeed supposed to be a union